src/input.jl

```diff
@@ -0,0 +1,96 @@
+
+struct GenericLinearObservation <: AbstractObservation
+end
+struct GenericLinearEvolution{TPhi,Tb,TQ} <: AbstractEvolution
+end
+
+"""
+    LinearEvolution(Phi, b, Q)
+
+Evolution `x -> Phi x + b + w` where ``w ~ N(0, Q)``
+"""
+struct LinearEvolution{TPhi,Tb,TQ} <: AbstractEvolution
+    Phi::TPhi # dxd
+    b::Tb # d
+    Q::TQ # dxd
+end
+
+"""
+    LinearObservation(H, R)
+
+Observe `y = Hx + v` where ``v ~ N(0, R)``
+"""
+struct LinearObservation{TH, TR} <: AbstractObservation
+    H::TH # dxd
+    R::TR # d
+end
+
+
+function predict!(s, x, P, t, u, M) 
+    if s == t # if `s == t`` no time has passed
+        x + u, P, one(M.Phi)
+    else
+        M.Phi*x + M.b + u, M.Phi*P*M.Phi' + M.Q, M.Phi
```

Comment:
I think I understand why you have both `M.b` and `u`, but the redundancy makes me a bit uncomfortable. They're the same quantity. Perhaps we should assert that `uu==nothing` when using `LinearEvolution`?

Reply:
How I think of them is that `b` is the intrinsic drift and `u` is an control/external signal.

Reply:
Perhaps, but unless I'm mistaken, `u` is unspecified (implicitly 0?) when `GenericLinearEvolution` is used.

Reply:
GenericLinearEvolution will dispatch on function https://github.com/mschauer/Kalman.jl/pull/5/files/0826886fe9698002c68cc8c16c3c8c39a6e8d92e#diff-d0bf3c72d95233c17cc7933aeaa30298R37

Reply:
Right. What I mean is that with `LinearObservation` (line 33), the evolution is `Phi*x + M.b + u` (transition matrix + intrinsic drift + control/external signal), whereas with `GenericLinearEvolution` (line 39), it's `Phi*x + b`

Anyway, it's no big deal; just seemed a bit inconsistent.